pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -156,6 +158,16 @@ func (cfg Config) toAuthOptions() gophercloud.AuthOptions {
 		// Persistent service, so we need to be able to renew tokens.
 		AllowReauth: true,
 	}
+
+	options.TenantID = cfg.Global.TenantId
+	options.TenantName = cfg.Global.TenantName
+	if cfg.Global.ProjectId != "" {
+		options.TenantID = cfg.Global.ProjectId
```

Comment:
Good opt. I also met this issue, but I specify user-id and trust-id to address it.
It is better to add project-id and project-name opts, but I can't understand why use the value of project-id(V3) for tenant-id(v2). 
Can we use tokens3.AuthOptions instead of gophercloud.AuthOptions when user specify project-id?
After checking gophercloud, it seems support project-id and project-name: https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/tokens/requests.go#L7